packages/data/src/onboarding/selectors.ts

```diff
@@ -0,0 +1,114 @@
+/**
+ * Internal dependencies
+ */
+import { WPDataSelectors } from '../types';
+
+export const getProfileItems = (
+	state: OnboardingState
+): ProfileItemsState | Record< string, never > => {
+	return state.profileItems || {};
+};
+
+export const getTasksStatus = (
+	state: OnboardingState
+): TasksStatusState | Record< string, never > => {
+	return state.tasksStatus || {};
+};
+
+export const getOnboardingError = (
+	state: OnboardingState,
+	selector: string
+): unknown | false => {
+	return state.errors[ selector ] || false;
+};
+
+export const isOnboardingRequesting = (
+	state: OnboardingState,
+	selector: string
+): boolean => {
+	return state.requesting[ selector ] || false;
+};
+
+// Types
+export type OnboardingSelectors = {
+	getProfileItems: () => ReturnType< typeof getProfileItems >;
+	getTasksStatus: () => ReturnType< typeof getTasksStatus >;
+	getOnboardingError: () => ReturnType< typeof getOnboardingError >;
+	isOnboardingRequesting: () => ReturnType< typeof isOnboardingRequesting >;
+} & WPDataSelectors;
+
+export type OnboardingState = {
+	profileItems: ProfileItemsState;
+	tasksStatus: TasksStatusState;
+	// TODO clarify what the error record's type is
+	errors: Record< string, unknown >;
+	requesting: Record< string, boolean >;
+};
+
+export type TasksStatusState = {
+	automatedTaxSupportedCountries: string[];
+	enabledPaymentGateways: string[];
+	hasHomepage: boolean;
+	hasPaymentGateway: boolean;
+	hasPhysicalProducts: boolean;
+	hasProducts: boolean;
+	isAppearanceComplete: boolean;
+	isTaxComplete: boolean;
+	shippingZonesCount: number;
+	stripeSupportedCountries: string[];
+	stylesheet: string;
+	taxJarActivated: boolean;
+	// TODO - fill out this type
+	themeMods: unknown;
+	wcPayIsConnected: false;
+};
+
+export type Industry = {
+	slug: string;
+};
+
+export type ProductCount = '0' | '1-10' | '11-100' | '101 - 1000' | '1000+';
+
+export type ProductTypeSlug =
+	| 'physical'
+	| 'bookings'
+	| 'download'
+	| 'memberships'
+	| 'product-add-ons'
+	| 'product-bundles'
+	| 'subscriptions';
+
+export type OtherPlatformSlug =
+	| 'shopify'
+	| 'bigcommerce'
+	| 'wix'
+	| 'amazon'
+	| 'ebay'
+	| 'etsy'
+	| 'squarespace'
+	| 'other';
+
+export type RevenueTypeSlug =
+	| 'none'
+	| 'rather-not-say'
+	| 'up-to-2500'
+	| '2500-10000'
+	| '10000-50000'
+	| '50000-250000'
+	| 'more-than-250000';
+
+export type ProfileItemsState = {
```

Comment:
Not sure what you have done in the past, but I tend to prefer to use `interface` for describing an object, I guess the difference is pretty subtle: https://www.typescriptlang.org/docs/handbook/advanced-types.html#interfaces-vs-type-aliases although we probably do want to decide where we draw the line :)

I guess I like what they said here ;) 

> Because an interface more closely maps how JavaScript objects work by being open to extension, we recommend using an interface over a type alias when possible.
>
> On the other hand, if you can’t express some shape with an interface and you need to use a union or tuple type, type aliases are usually the way to go.

Reply:
@louwie17 yeah I'm aware of the difference. Although its true you can extend a JS object, having a closed type discourages this. It's also shorter and more succinct so I prefer it because the 2 are so close to being the same thing.

You're looking at the old docs though, this is the new docs: https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces

They say:

"For the most part, you can choose based on personal preference, and TypeScript will tell you if it needs something to be the other kind of declaration. If you would like a heuristic, use interface until you need to use features from type."

So my feeling is for something like `Window` absolutely it should be an interface, especially because it's probably going to get extended many times, but for a type definition I want to remain closed to extension `type` makes more sense to me.

Reply:
@louwie17 I'm usually one for consistency but frankly it doesn't bother me too much what you use, if you prefer interface I say carry on. The docs are clear that it *almost never* matters. I figured for these package types I may as well keep them closed, I really hope no one extends them, but for many other situations `interface` will be just fine. 

I wonder if maybe we should consider using `interface` in client more exclusively, and `type` in packages? I'm not sure. There are some situations where the errors delivered about `interface` are supposedly clearer according to the docs, but I hadn't ever found that an issue.

Reply:
I'm gonna leave this for now. We could later decide on some heuristics/best practices for the team with TS, but I think its ok to use either.